packages/cli/generators/service/index.js

```diff
@@ -233,6 +233,12 @@ module.exports = class ServiceGenerator extends ArtifactGenerator {
           this.artifactInfo.dataSourceClass,
         );
 
+        if (!this.artifactInfo.dataSourceName) {
+          return Promise.reject(
+            'Error: datasource config does not have `name` property',
+          );
+        }
```

Comment:
It's probably simpler to just `throw new Error('Datasource config does not have `name` property'');`.